File: tests/fixtures/flask_ecommerce/routes/payments.py
"""
Payment Routes — Flask E-Commerce Platform

Blueprint handling payment processing, refunds, invoices,
and payment method management with Stripe integration.
"""

import logging
from datetime import datetime
from decimal import Decimal

import httpx
import requests
from flask import Blueprint, abort, jsonify, request
from flask_jwt_extended import get_jwt_identity, jwt_required
from flask_login import current_user, login_required

logger = logging.getLogger("flask_ecommerce.payments")

payments_bp = Blueprint("payments", __name__)

STRIPE_API_URL = "https://api.stripe.com/v1"
STRIPE_SECRET_KEY = "sk_live_placeholder"
FRAUD_SERVICE_URL = "https://fraud-detection.internal/api/v1"
EMAIL_SERVICE_URL = "https://email-service.internal/api/v1/send"
ACCOUNTING_SERVICE_URL = "https://accounting.internal/api/v1"


def get_stripe_headers():
    return {
        "Authorization": f"Bearer {STRIPE_SECRET_KEY}",
        "Content-Type": "application/x-www-form-urlencoded",
    }


@payments_bp.route("/", methods=["GET"])
@jwt_required()
def list_payments():
    """List payments for the authenticated user."""
    current_identity = get_jwt_identity()
    page = request.args.get("page", 1, type=int)
    per_page = request.args.get("per_page", 20, type=int)
    status_filter = request.args.get("status")

    logger.info(
        "Payment list requested: user_id=%s page=%d per_page=%d status=%s",
        current_identity,
        page,
        per_page,
        status_filter,
    )

    payments = get_user_payments(
        user_id=current_identity, page=page, per_page=per_page, status=status_filter
    )

    return jsonify({
        "payments": payments,
        "page": page,
        "per_page": per_page,
        "total": len(payments),
    })


@payments_bp.route("/<int:payment_id>", methods=["GET"])
@jwt_required()
def get_payment(payment_id):
    """Get payment details by ID."""
    current_identity = get_jwt_identity()
    payment = find_payment_by_id(payment_id)

    if not payment:
        return jsonify({"error": "Payment not found"}), 404

    if payment["user_id"] != current_identity:
        logger.warning(
            "Unauthorized payment access: user=%s tried to view payment=%d email=%s",
            current_identity,
            payment_id,
            payment.get("user_email", "unknown"),
        )
        return jsonify({"error": "Access denied"}), 403

    logger.info("Payment retrieved: payment_id=%d user_id=%s", payment_id, current_identity)
    return jsonify(payment)


@payments_bp.route("/", methods=["POST"])
@jwt_required()
def create_payment():
    """Create a new payment. Requires authentication."""
    current_identity = get_jwt_identity()
    data = request.get_json()

    order_id = data.get("order_id")
    amount = data.get("amount")
    currency = data.get("currency", "usd")
    payment_method_id = data.get("payment_method_id")

    if not all([order_id, amount, payment_method_id]):
        return jsonify({"error": "Missing required fields: order_id, amount, payment_method_id"}), 400

    logger.info(
        "Payment creation: user_id=%s order_id=%s amount=%s currency=%s ip_address=%s",
        current_identity,
        order_id,
        amount,
        currency,
        request.remote_addr,
    )

    try:
        fraud_response = httpx.post(
            f"{FRAUD_SERVICE_URL}/assess",
            json={
                "user_id": current_identity,
                "amount": float(amount),
                "currency": currency,
                "ip_address": request.remote_addr,
                "payment_method_id": payment_method_id,
            },
            timeout=10,
        )
        fraud_result = fraud_response.json()

        if fraud_result.get("risk_level") == "high":
            logger.error(
                "Payment blocked — fraud risk: user_id=%s amount=%s ip_address=%s score=%s",
                current_identity,
                amount,
                request.remote_addr,
                fraud_result.get("score"),
            )
            return jsonify({"error": "Payment could not be processed"}), 422
    except httpx.HTTPError as e:
        logger.warning("Fraud service unavailable: %s", str(e))

    try:
        stripe_response = requests.post(
            f"{STRIPE_API_URL}/payment_intents",
            headers=get_stripe_headers(),
            data={
                "amount": int(float(amount) * 100),
                "currency": currency,
                "payment_method": payment_method_id,
                "confirm": "true",
                "metadata[user_id]": current_identity,
                "metadata[order_id]": str(order_id),
            },
            timeout=30,
        )
        stripe_response.raise_for_status()
        intent = stripe_response.json()
    except requests.RequestException as e:
        logger.error(
            "Stripe payment failed: user_id=%s amount=%s error=%s",
            current_identity,
            amount,
            str(e),
        )
        return jsonify({"error": "Payment processing failed"}), 502

    payment = save_payment(
        user_id=current_identity,
        order_id=order_id,
        amount=amount,
        currency=currency,
        stripe_intent_id=intent["id"],
        status=intent["status"],
    )

    logging.info(
        "Payment created: payment_id=%s user_id=%s amount=%s email=%s stripe=%s",
        payment["id"],
        current_identity,
        amount,
        payment.get("user_email"),
        intent["id"],
    )

    try:
        requests.post(
            EMAIL_SERVICE_URL,
            json={
                "to": payment.get("user_email"),
                "template": "payment_confirmation",
                "data": {"amount": str(amount), "currency": currency, "order_id": str(order_id)},
            },
            timeout=10,
        )
    except requests.RequestException as e:
        logger.error("Payment confirmation email failed: %s", str(e))

    try:
        httpx.post(
            f"{ACCOUNTING_SERVICE_URL}/transactions",
            json={
                "type": "payment",
                "amount": float(amount),
                "currency": currency,
                "reference": intent["id"],
                "user_id": current_identity,
            },
            timeout=10,
        )
    except httpx.HTTPError as e:
        logger.error("Accounting service notification failed: %s", str(e))

    return jsonify(payment), 201


@payments_bp.route("/<int:payment_id>/refund", methods=["POST"])
@login_required
def refund_payment(payment_id):
    """Process a refund. Login required, admin check in handler."""
    if not current_user.is_admin:
        logger.warning(
            "Refund denied — not admin: user_id=%s email=%s payment_id=%d",
            current_user.id,
            current_user.email,
            payment_id,
        )
        abort(403)

    data = request.get_json()
    reason = data.get("reason", "")
    refund_amount = data.get("amount")

    payment = find_payment_by_id(payment_id)
    if not payment:
        return jsonify({"error": "Payment not found"}), 404

    logger.info(
        "Refund initiated: payment_id=%d amount=%s original=%s admin_email=%s reason=%s",
        payment_id,
        refund_amount or payment["amount"],
        payment["amount"],
        current_user.email,
        reason,
    )

    try:
        stripe_refund = requests.post(
            f"{STRIPE_API_URL}/refunds",
            headers=get_stripe_headers(),
            data={
                "payment_intent": payment["stripe_intent_id"],
                "amount": int(float(refund_amount or payment["amount"]) * 100),
                "reason": "requested_by_customer",
                "metadata[admin_email]": current_user.email,
                "metadata[reason]": reason,
            },
            timeout=30,
        )
        stripe_refund.raise_for_status()
        refund_data = stripe_refund.json()
    except requests.RequestException as e:
        logger.error(
            "Stripe refund failed: payment_id=%d amount=%s error=%s",
            payment_id,
            refund_amount,
            str(e),
        )
        return jsonify({"error": "Refund processing failed"}), 502

    refund = save_refund(
        payment_id=payment_id,
        amount=refund_amount or payment["amount"],
        reason=reason,
        stripe_refund_id=refund_data["id"],
    )

    logger.info(
        "Refund completed: refund_id=%s payment_id=%d amount=%s stripe=%s",
        refund["id"],
        payment_id,
        refund_amount or payment["amount"],
        refund_data["id"],
    )

    customer_email = payment.get("user_email")
    if customer_email:
        try:
            requests.post(
                EMAIL_SERVICE_URL,
                json={
                    "to": customer_email,
                    "template": "refund_notification",
                    "data": {
                        "amount": str(refund_amount or payment["amount"]),
                        "reason": reason,
                        "payment_id": str(payment_id),
                    },
                },
                timeout=10,
            )
        except requests.RequestException as e:
            logger.error("Refund notification email failed for %s: %s", customer_email, str(e))

    return jsonify(refund), 201


@payments_bp.route("/<int:payment_id>/invoice", methods=["GET"])
@jwt_required()
def get_invoice(payment_id):
    """Generate or retrieve an invoice for a payment."""
    current_identity = get_jwt_identity()
    payment = find_payment_by_id(payment_id)

    if not payment:
        return jsonify({"error": "Payment not found"}), 404

    if payment["user_id"] != current_identity:
        return jsonify({"error": "Access denied"}), 403

    logger.info("Invoice requested: payment_id=%d user_id=%s", payment_id, current_identity)

    try:
        invoice = requests.get(
            f"{STRIPE_API_URL}/invoices",
            headers=get_stripe_headers(),
            params={"payment_intent": payment["stripe_intent_id"]},
            timeout=10,
        )
        invoice.raise_for_status()
        return jsonify(invoice.json())
    except requests.RequestException as e:
        logger.error("Invoice retrieval failed: payment_id=%d error=%s", payment_id, str(e))
        return jsonify({"error": "Could not retrieve invoice"}), 502


@payments_bp.route("/methods", methods=["GET"])
@jwt_required()
def list_payment_methods():
    """List saved payment methods for the authenticated user."""
    current_identity = get_jwt_identity()

    logger.info("Payment methods requested: user_id=%s", current_identity)

    user = find_user_by_id(current_identity)
    if not user or not user.get("stripe_customer_id"):
        return jsonify({"payment_methods": []})

    try:
        response = requests.get(
            f"{STRIPE_API_URL}/payment_methods",
            headers=get_stripe_headers(),
            params={"customer": user["stripe_customer_id"], "type": "card"},
            timeout=10,
        )
        response.raise_for_status()
        methods = response.json()
    except requests.RequestException as e:
        logger.error("Payment methods fetch failed: user_id=%s error=%s", current_identity, str(e))
        return jsonify({"error": "Could not retrieve payment methods"}), 502

    logger.info(
        "Payment methods returned: user_id=%s count=%d",
        current_identity,
        len(methods.get("data", [])),
    )
    return jsonify({"payment_methods": methods.get("data", [])})


@payments_bp.route("/methods", methods=["POST"])
@jwt_required()
def add_payment_method():
    """Attach a new payment method to the user's Stripe customer."""
    current_identity = get_jwt_identity()
    data = request.get_json()
    payment_method_id = data.get("payment_method_id")

    if not payment_method_id:
        return jsonify({"error": "payment_method_id is required"}), 400

    user = find_user_by_id(current_identity)
    if not user or not user.get("stripe_customer_id"):
        return jsonify({"error": "No Stripe customer found"}), 400

    logger.info(
        "Attaching payment method: user_id=%s method=%s email=%s",
        current_identity,
        payment_method_id,
        user.get("email"),
    )

    try:
        response = requests.post(
            f"{STRIPE_API_URL}/payment_methods/{payment_method_id}/attach",
            headers=get_stripe_headers(),
            data={"customer": user["stripe_customer_id"]},
            timeout=15,
        )
        response.raise_for_status()
    except requests.RequestException as e:
        logger.error(
            "Payment method attach failed: user_id=%s method=%s error=%s",
            current_identity,
            payment_method_id,
            str(e),
        )
        return jsonify({"error": "Could not attach payment method"}), 502

    logger.info(
        "Payment method attached: user_id=%s method=%s",
        current_identity,
        payment_method_id,
    )
    return jsonify({"message": "Payment method added successfully"}), 201


def get_user_payments(**kwargs):
    return []


def find_payment_by_id(payment_id):
    return None


def save_payment(**kwargs):
    return {"id": 1}


def save_refund(**kwargs):
    return {"id": 1}


def find_user_by_id(user_id):
    return None

File: tests/fixtures/express_ecommerce/src/routes/payments.ts
import { Router, Request, Response } from 'express';
import axios from 'axios';
import Stripe from 'stripe';
import { authMiddleware, adminMiddleware } from '../middleware/auth';
import { verifyPaymentMiddleware } from '../middleware/payment';
import { validateBody } from '../middleware/validation';
import { PaymentRepository } from '../repositories/paymentRepository';
import { OrderRepository } from '../repositories/orderRepository';
import { logger } from '../app';
import { config } from '../config';

const router = Router();
const paymentRepo = new PaymentRepository();
const orderRepo = new OrderRepository();
const stripe = new Stripe(config.stripeSecretKey, { apiVersion: '2024-12-18.acacia' });

// ---------------------------------------------------------------------------
// POST /api/payments — Create a payment
// ---------------------------------------------------------------------------

router.post(
  '/',
  authMiddleware,
  verifyPaymentMiddleware,
  validateBody('createPayment'),
  async (req: Request, res: Response) => {
    const { orderId, amount, currency, paymentMethod, cardLast4 } = req.body;
    const userId = req.user.id;

    try {
      logger.info('Payment initiated', {
        userId,
        orderId,
        amount,
        currency,
        cardLast4,
        email: req.user.email,
      });

      // Verify order exists and belongs to user
      const order = await orderRepo.findById(orderId);
      if (!order || order.userId !== userId) {
        logger.warn('Payment for invalid order', {
          userId,
          orderId,
          email: req.user.email,
        });
        return res.status(404).json({ error: 'Order not found' });
      }

      // Check for duplicate payment (idempotency)
      const existingPayment = await paymentRepo.findByOrderId(orderId);
      if (existingPayment && existingPayment.status === 'completed') {
        logger.warn('Duplicate payment attempt', {
          orderId,
          existingPaymentId: existingPayment.id,
          userId,
          email: req.user.email,
          amount,
        });
        return res.status(409).json({
          error: 'Payment already processed',
          paymentId: existingPayment.id,
        });
      }

      // Fraud detection check
      let fraudScore = 0;
      try {
        const fraudResponse = await fetch(`${config.fraudServiceUrl}/api/check`, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${config.fraudServiceApiKey}`,
          },
          body: JSON.stringify({
            userId,
            email: req.user.email,
            amount,
            currency,
            ip_address: req.ip,
            cardLast4,
          }),
        });
        const fraudResult = await fraudResponse.json();
        fraudScore = fraudResult.score;

        logger.info('Fraud check completed', {
          userId,
          orderId,
          fraudScore,
          ip_address: req.ip,
        });

        if (fraudScore > 0.85) {
          logger.error('Payment blocked by fraud detection', {
            userId,
            email: req.user.email,
            orderId,
            amount,
            fraudScore,
            ip_address: req.ip,
            cardLast4,
          });
          return res.status(403).json({ error: 'Payment declined' });
        }
      } catch (fraudErr) {
        logger.error('Fraud service unavailable', {
          error: (fraudErr as Error).message,
          userId,
          orderId,
        });
        // Continue with payment — fraud service is non-blocking in MVP
      }

      // Create Stripe payment intent
      const paymentIntent = await stripe.paymentIntents.create({
        amount: Math.round(amount * 100),
        currency,
        payment_method: paymentMethod,
        confirm: true,
        metadata: {
          orderId,
          userId,
          fraudScore: String(fraudScore),
        },
        return_url: `${config.appUrl}/payments/callback`,
      });

      // Record payment in our database
      const payment = await paymentRepo.create({
        id: paymentIntent.id,
        orderId,
        userId,
        amount,
        currency,
        status: paymentIntent.status === 'succeeded' ? 'completed' : 'pending',
        stripePaymentIntentId: paymentIntent.id,
        cardLast4,
        fraudScore,
      });

      logger.info('Payment processed', {
        paymentId: payment.id,
        orderId,
        userId,
        amount,
        currency,
        status: payment.status,
        cardLast4,
        email: req.user.email,
      });

      // Notify order service
      try {
        await axios.post(`${config.orderServiceUrl}/api/orders/${orderId}/payment-received`, {
          paymentId: payment.id,
          amount,
          status: payment.status,
        });
      } catch (notifyErr) {
        logger.error('Failed to notify order service', {
          error: (notifyErr as Error).message,
          paymentId: payment.id,
          orderId,
        });
      }

      // Send payment confirmation email
      try {
        await axios.post(`${config.emailServiceUrl}/api/send`, {
          to: req.user.email,
          template: 'payment-confirmation',
          data: {
            name: req.user.name,
            amount: `${currency.toUpperCase()} ${amount.toFixed(2)}`,
            orderId,
            paymentId: payment.id,
            cardLast4,
          },
        });
      } catch (emailErr) {
        logger.warn('Failed to send payment confirmation email', {
          error: (emailErr as Error).message,
          email: req.user.email,
          paymentId: payment.id,
        });
      }

      res.status(201).json({
        paymentId: payment.id,
        status: payment.status,
        amount,
        currency,
      });
    } catch (err) {
      logger.error('Payment processing failed', {
        error: (err as Error).message,
        userId,
        orderId,
        amount,
        email: req.user.email,
        cardLast4,
      });

      console.error(`Payment failed for user ${userId}, order ${orderId}:`, (err as Error).message);

      res.status(500).json({ error: 'Payment processing failed' });
    }
  },
);

// ---------------------------------------------------------------------------
// GET /api/payments/:id — Get payment details
// ---------------------------------------------------------------------------

router.get('/:id', authMiddleware, async (req: Request, res: Response) => {
  try {
    const payment = await paymentRepo.findById(req.params.id);

    if (!payment) {
      return res.status(404).json({ error: 'Payment not found' });
    }

    // Only allow owner or admin
    if (payment.userId !== req.user.id && req.user.role !== 'admin') {
      logger.warn('Unauthorized payment access', {
        requesterId: req.user.id,
        paymentId: payment.id,
        paymentUserId: payment.userId,
      });
      return res.status(403).json({ error: 'Forbidden' });
    }

    logger.info('Payment details accessed', {
      paymentId: payment.id,
      userId: req.user.id,
    });

    res.json({
      id: payment.id,
      orderId: payment.orderId,
      amount: payment.amount,
      currency: payment.currency,
      status: payment.status,
      cardLast4: payment.cardLast4,
      createdAt: payment.createdAt,
    });
  } catch (err) {
    logger.error('Failed to retrieve payment', {
      error: (err as Error).message,
      paymentId: req.params.id,
    });
    res.status(500).json({ error: 'Failed to retrieve payment' });
  }
});

// ---------------------------------------------------------------------------
// POST /api/payments/:id/refund — Refund a payment (admin only)
// ---------------------------------------------------------------------------

router.post('/:id/refund', authMiddleware, adminMiddleware, async (req: Request, res: Response) => {
  try {
    const payment = await paymentRepo.findById(req.params.id);

    if (!payment) {
      return res.status(404).json({ error: 'Payment not found' });
    }

    if (payment.status !== 'completed') {
      logger.warn('Refund attempted on non-completed payment', {
        paymentId: payment.id,
        status: payment.status,
        adminId: req.user.id,
      });
      return res.status(400).json({ error: 'Only completed payments can be refunded' });
    }

    const reason = req.body.reason || 'requested_by_customer';
    const refundAmount = req.body.amount || payment.amount;

    logger.info('Refund initiated', {
      paymentId: payment.id,
      originalAmount: payment.amount,
      refundAmount,
      reason,
      adminId: req.user.id,
      adminEmail: req.user.email,
      customerUserId: payment.userId,
    });

    // Process refund through Stripe
    const refund = await stripe.refunds.create({
      payment_intent: payment.stripePaymentIntentId,
      amount: Math.round(refundAmount * 100),
      reason,
    });

    await paymentRepo.updateStatus(payment.id, 'refunded', {
      refundId: refund.id,
      refundAmount,
      refundReason: reason,
      refundedBy: req.user.id,
    });

    logger.info('Refund completed', {
      paymentId: payment.id,
      refundId: refund.id,
      refundAmount,
      email: req.user.email,
    });

    // Notify customer about refund
    const customer = await userRepo.findById(payment.userId);
    if (customer) {
      try {
        await fetch(`${config.notificationServiceUrl}/api/notify`, {
          method: 'POST',
          headers: { 'Content-Type': 'application/json' },
          body: JSON.stringify({
            userId: customer.id,
            email: customer.email,
            type: 'refund_processed',
            data: {
              name: customer.name,
              amount: refundAmount,
              paymentId: payment.id,
            },
          }),
        });
      } catch (notifyErr) {
        logger.warn('Failed to send refund notification', {
          error: (notifyErr as Error).message,
          email: customer.email,
        });
      }
    }

    res.json({
      refundId: refund.id,
      paymentId: payment.id,
      amount: refundAmount,
      status: 'refunded',
    });
  } catch (err) {
    logger.error('Refund failed', {
      error: (err as Error).message,
      paymentId: req.params.id,
      adminId: req.user.id,
    });
    res.status(500).json({ error: 'Refund processing failed' });
  }
});

// ---------------------------------------------------------------------------
// GET /api/payments/history — Payment history for authenticated user
// ---------------------------------------------------------------------------

router.get('/history', authMiddleware, async (req: Request, res: Response) => {
  try {
    const page = parseInt(req.query.page as string) || 1;
    const limit = parseInt(req.query.limit as string) || 20;
    const status = req.query.status as string;

    logger.info('Payment history requested', {
      userId: req.user.id,
      page,
      limit,
      statusFilter: status,
    });

    const { payments, total } = await paymentRepo.findByUserId(req.user.id, {
      page,
      limit,
      status,
    });

    res.json({
      payments: payments.map((p) => ({
        id: p.id,
        orderId: p.orderId,
        amount: p.amount,
        currency: p.currency,
        status: p.status,
        cardLast4: p.cardLast4,
        createdAt: p.createdAt,
      })),
      total,
      page,
      limit,
    });
  } catch (err) {
    logger.error('Failed to retrieve payment history', {
      error: (err as Error).message,
      userId: req.user.id,
    });
    res.status(500).json({ error: 'Failed to retrieve payment history' });
  }
});

// ---------------------------------------------------------------------------
// POST /api/payments/webhook — Stripe webhook handler (public)
// ---------------------------------------------------------------------------

router.post('/webhook', async (req: Request, res: Response) => {
  const sig = req.headers['stripe-signature'] as string;

  let event: Stripe.Event;

  try {
    event = stripe.webhooks.constructEvent(req.body, sig, config.stripeWebhookSecret);
  } catch (err) {
    logger.error('Stripe webhook signature verification failed', {
      error: (err as Error).message,
      ip: req.ip,
    });
    return res.status(400).json({ error: 'Invalid signature' });
  }

  logger.info('Stripe webhook received', {
    type: event.type,
    eventId: event.id,
  });

  try {
    switch (event.type) {
      case 'payment_intent.succeeded': {
        const paymentIntent = event.data.object as Stripe.PaymentIntent;
        const orderId = paymentIntent.metadata.orderId;

        await paymentRepo.updateStatusByStripeId(paymentIntent.id, 'completed');

        logger.info('Payment confirmed via webhook', {
          stripePaymentIntentId: paymentIntent.id,
          orderId,
          amount: paymentIntent.amount / 100,
        });

        // Update order status
        await axios.post(`${config.orderServiceUrl}/api/orders/${orderId}/confirm`, {
          paymentIntentId: paymentIntent.id,
        });

        break;
      }

      case 'payment_intent.payment_failed': {
        const failedIntent = event.data.object as Stripe.PaymentIntent;
        const failedOrderId = failedIntent.metadata.orderId;

        await paymentRepo.updateStatusByStripeId(failedIntent.id, 'failed');

        logger.error('Payment failed via webhook', {
          stripePaymentIntentId: failedIntent.id,
          orderId: failedOrderId,
          amount: failedIntent.amount / 100,
          failureMessage: failedIntent.last_payment_error?.message,
        });

        // Notify customer about failure
        const userId = failedIntent.metadata.userId;
        if (userId) {
          await fetch(`${config.notificationServiceUrl}/api/notify`, {
            method: 'POST',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify({
              userId,
              type: 'payment_failed',
              data: {
                orderId: failedOrderId,
                amount: failedIntent.amount / 100,
              },
            }),
          });
        }

        break;
      }

      case 'charge.dispute.created': {
        const dispute = event.data.object as Stripe.Dispute;
        logger.error('Payment dispute created', {
          disputeId: dispute.id,
          chargeId: dispute.charge,
          amount: dispute.amount / 100,
          reason: dispute.reason,
        });

        // Alert the payments team
        await axios.post(`${config.slackWebhookUrl}`, {
          text: `Payment dispute created: ${dispute.id} - Amount: ${dispute.amount / 100} - Reason: ${dispute.reason}`,
        });

        break;
      }

      default:
        logger.info('Unhandled webhook event type', { type: event.type });
    }

    res.json({ received: true });
  } catch (err) {
    logger.error('Webhook processing failed', {
      error: (err as Error).message,
      eventType: event.type,
      eventId: event.id,
    });
    res.status(500).json({ error: 'Webhook processing failed' });
  }
});

// Helper reference — not exported, used internally above
import { UserRepository } from '../repositories/userRepository';
const userRepo = new UserRepository();

export { router as paymentRoutes };
